crates/doctor/src/rustlang/target.rs

```diff
@@ -0,0 +1,192 @@
+        Some(self)
+    }
+}
+
+#[async_trait]
+impl RustTreatment for TargetDiagnosis {
+    fn summary(&self) -> String {
+        match self {
+            Self::RustNotInstalled => "Install the Rust compiler and the wasm32-wasi target",
+            Self::WasmTargetNotInstalled => "Install the Rust wasm32-wasi target",
+        }
+        .into()
+    }
+
+    async fn treat_rust(&self) -> anyhow::Result<()> {
+        match self {
+            Self::RustNotInstalled => {
+                install_rust().await?;
+                install_wasi_target()?;
+            }
+            Self::WasmTargetNotInstalled => {
+                install_wasi_target()?;
+            }
+        }
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Treatment for TargetDiagnosis {
+    fn summary(&self) -> String {
+        RustTreatment::summary(self)
+    }
+
+    async fn dry_run(&self, _patient: &PatientApp) -> Result<String> {
+        let message = match self {
+            Self::RustNotInstalled => "Download and run the Rust installer from https://rustup.rs, then run `rustup target add wasm32-wasi`",
+            Self::WasmTargetNotInstalled => "Run the following command:\n    `rustup target add wasm32-wasi`",
+        };
+        Ok(message.into())
+    }
+
+    async fn treat(&self, _patient: &mut PatientApp) -> Result<()> {
+        self.treat_rust().await
+    }
+}
+
+async fn install_rust() -> anyhow::Result<()> {
+    let status = run_rust_installer().await?;
+    anyhow::ensure!(status.success(), "Rust installation failed: {status:?}");
+    Ok(())
+}
+
+#[cfg(not(windows))]
+async fn run_rust_installer() -> anyhow::Result<std::process::ExitStatus> {
+    use std::io::Write;
+
+    let resp = reqwest::get("https://sh.rustup.rs").await?;
+    let script = resp.bytes().await?;
+
+    let mut cmd = std::process::Command::new("sh");
+    cmd.stdin(std::process::Stdio::piped());
+    let mut shell = cmd.spawn()?;
+    let mut stdin = shell.stdin.take().unwrap();
+    std::thread::spawn(move || {
+        stdin.write_all(&script).unwrap();
+    });
+
+    let output = shell.wait_with_output()?;
+    Ok(output.status)
+}
+
+#[cfg(windows)]
+async fn run_rust_installer() -> anyhow::Result<std::process::ExitStatus> {
+    // We currently distribute Windows builds only for x64, so hopefully
+    // this won't be an issue.
+    if std::env::consts::ARCH != "x86_64" {
+        anyhow::bail!("Spin Doctor can only install Rust for Windows on x64 processors");
+    }
+
+    let temp_dir = tempfile::TempDir::new()?;
+    let installer_path = temp_dir.path().join("rustup-init.exe");
+
+    let resp = reqwest::get("https://win.rustup.rs/x86_64").await?;
+    let installer_bin = resp.bytes().await?;
+
+    std::fs::write(&installer_path, &installer_bin)?;
+
+    let mut cmd = std::process::Command::new(installer_path);
+    let status = cmd.status()?;
+    Ok(status)
+}
+
+fn install_wasi_target() -> anyhow::Result<()> {
```

Comment:
```suggestion
fn install_wasi_target() -> Result<()> {
```

Reply:
I wish we actually kept it as `anyhow::Result` across the entire code base... 🥲

Reply:
Personally, I agree. As long as we are consistent.